Elegoo Neptune Boards

mlee12382 opened the pull request:
<!--

Submitting a Pull Request

- Please fill out all sections of this form. You can delete the helpful comments.
- Pull Requests without clear information will take longer and may even be rejected.
- We get a high volume of submissions so please be patient during review.

-->

### Description
Added new board definitions and pins and updated .ini files for platformio for the Elegoo Neptune series of printers.
<!--

Clearly describe the submitted changes with lots of details. Include images where helpful. Initial reviewers may not be familiar with the subject, so be as thorough as possible. You can use MarkDown syntax to improve readability with bullet lists, code blocks, and so on. PREVIEW and fix up formatting before submitting.

-->

### Requirements
N/A
<!-- Does this PR require a specific board, LCD, etc.? -->

### Benefits
Adds boards to make custom compiling firmware for the Elegoo Neptune series of printers easier, duplicates and renames the 2 boards that come in the Neptune 2 printers to make it more straightforward and adds the missing board files for the Neptune X and Neptune 3
<!-- What does this PR fix or improve? -->

### Configurations
N/A
<!-- Attach Configurations ZIP and any other files needed to test this PR. -->

### Related Issues
N/A
<!-- Does this PR fix a bug or fulfill a Feature Request? Link related Issues here. -->

mlee12382 committed:
Added Elegoo Boards

Added Elegoo Neptune series boards. ZNP Robin Nano v1.2 & v1.3, Neptune X, and Neptune 3.

mlee12382 committed:
Added Elegoo Boards

Added Elegoo Neptune Board definitions, ZNP_ROBIN_NANO_V1_3, ZNP_ROBIN_NANO, MKS_NEPTUNE_X, and MKS_NEPTUNE_3

mlee12382 committed:
Added Elegoo Neptune pins

Added definitions for Elegoo Neptune board pins files. ZNP_ROBIN_NANO, ZNP_ROBIN_NANO_V1_3, MKS_NEPTUNE_X, and MKS_NEPTUNE_3

mlee12382 committed:
Added Elegoo Neptune Boards

Added pins file for Elegoo Neptune boards, pins_MKS_NEPTUNE_X.h, pins_MKS_NEPTUNE_X_common.h, and pins_NEPTUNE_3.h

mlee12382 committed:
Merge branch 'MarlinFirmware:bugfix-2.1.x' into elegoo-neptune-3

mlee12382 committed:
Added Elegoo Neptune Board

Added Elegoo Neptune 2 board, znp_robin_nano35

mlee12382 committed:
Merge branch 'MarlinFirmware:bugfix-2.1.x' into elegoo-neptune-3

mlee12382 committed:
Added Elegoo Neptune Boards

Added environments for Elegoo Neptune Series printers, mks_neptune_x, mks_neptune_3, and znp_robin_nano_v1_3

mlee12382 committed:
Added Elegoo Neptune 2 board

Added Elegoo Neptune 2 v.12 board, znp_robin_nano35, duplicates mks_robin_nano35 but changes output file name and makes it more straightforward to compile.

EvilGremlin commented:
Redundant, ZNP_ROBIN_NANO exactly same as MKS_ROBIN_NANO

mlee12382 commented:
> Redundant, ZNP_ROBIN_NANO is exactly same as MKS_ROBIN_NANO

The board is a clone of the MKS but it is labeled ZNP on the control board, the redundancy is to avoid confusion for people wanting to build their own custom firmware.

EvilGremlin commented:
MKS Robin (nano) E3 also exist already. 
Which leaves quiestion: is Neptune X board unique or also rebadge of something else?

To avoid confusion simple name alias is more than enough, no need to make a copy of pin files
#elif MB(ZNP_ROBIN_NANO)
  #include "stm32f1/pins_MKS_ROBIN_NANO.h"
like that

mlee12382 commented:
> MKS Robin (nano) E3 also exist already. Which leaves quiestion: is Neptune X board unique or also rebadge of something else?
> 
> To avoid confusion simple name alias is more than enough, no need to make a copy of pin files #elif MB(ZNP_ROBIN_NANO) #include "stm32f1/pins_MKS_ROBIN_NANO.h" like that

Neptune X is unique as is Neptune 3, I think that's how I did the znp robin boards was just as aliases in boards.h

EvilGremlin commented on Marlin/src/pins/stm32f4/pins_NEPTUNE_3.h:
```diff
@@ -0,0 +1,139 @@
+//
+// #define DISABLE_DEBUG
+
+#define BOARD_INFO_NAME "MKS Robin Nano E3"
```
Yep, this is mistake, not nano e3 indeed

mlee12382 replied:
That was provided to me by elegoo, they based it on the nano e3 but it is custom

mlee12382 replied:
I just updated the name to fix the confusion

mlee12382 committed:
Board name change

Board Name change

mlee12382 committed:
Update pins_NEPTUNE_3.h

EvilGremlin commented on ini/stm32f1.ini:
```diff
@@ -233,6 +233,24 @@ build_unflags               = ${stm32_variant.build_unflags}
+#
+# ZNP Robin Nano V1.2
+#
+[env:znp_robin_nano35]
```
also no need for full copy, just set 
`extends = mks_robin_nano35
board_build.encrypt_mks = elegoo.bin`
also can be `#define FIRMWARE_BIN elegoo.bin` in example config

mlee12382 replied:
*facepalm* idk why I didn't think of that.. lol

EvilGremlin commented:
Do you have board photos? Google is silent about new ones

mlee12382 committed:
cleaned up files

cleaned up definition for znp_robin_nano_v1_3

mlee12382 committed:
cleaned up file

cleaned up znp_robin_nano35 definition

mlee12382 commented:
Only these from the user guides for the Neptune X and Neptune 3, do you need the Neptune 2 boards also?
![NEPTUNE X](https://user-images.githubusercontent.com/89716126/181865554-c655165d-8d1c-4ea9-9473-2dacf81c851e.png)
![NEPTUNE 3](https://user-images.githubusercontent.com/89716126/181865555-84ed92b1-54f2-421f-a8ef-e501f136aa0c.png)

EvilGremlin commented:
Neptune X is robin nano 1.3...

mlee12382 commented:
> Neptune X is robin nano 1.3...

No the pins don't match anything currently in marlin.
![Neptune-X](https://user-images.githubusercontent.com/89716126/181865997-09a765be-7967-4669-8637-46caaab588a6.png)

mlee12382 commented:
It's close but not 100% match

EvilGremlin commented:
Yep, i see... a bit of a mess. Do you have actual hardware to verify pinouts? Bacause this neptune X v1.0 photo clearly don't match pins.

mlee12382 commented:
> Yep, i see... a bit of a mess. Do you have actual hardware to verify pinouts? Bacause this neptune X v1.0 photo clearly don't match pins.

No unfortunately not :( I am trying to help out the community cause we've been lacking firmware other than stock for the X and the 3 is brand new. I may need to do a bit more research it seems.

mlee12382 commented:
> Neptune X is robin nano 1.3... defines USES_DIAG_JUMPERS and CS pins etc but photo says there are none

I copied the nano_v3 and modified the pins since that is what elegoo said was the board they cloned and modified, I can remove that definition if it's not relevant.

EvilGremlin commented:
Nah, let's leave it as is for now. Maybe someone with hardware will show up to verify.

mlee12382 commented:
> Nah, let's leave it as is for now. Maybe someone with hardware will show up to verify.

I just noticed that the cs pins interfere with the fsmc pins, not sure why I didn't catch that before, will fix it real quick.

mlee12382 committed:
Removed CS pin definitions

Removed cs pin definitions to resolve conflict with fsmc pins

thinkyhead committed:
line endings?

thinkyhead committed:
cleanup

thinkyhead commented:
Does this need additional verification or should it be merged in its current state?

EvilGremlin commented:
I print with this conf for a week (no bltouch though, might have missed something there in revamp)

EvilGremlin commented:
oops, wrong thread :)

mlee12382 commented:
> Does this need additional verification or should it be merged in its current state?
 

I need to do more work on this before it's ready for merging unfortunately  :(

NotLordCalvert commented:
I realize this is closed, but would a picture of the board help here? If not, what do you need @mlee12382?

mlee12382 commented:
> I realize this is closed, but would a picture of the board help here? If not, what do you need @mlee12382?

For the Neptune 3 I need to figure out why it is having trouble compiling and why the SanityCheck is convinced that E0_AUTO_FAN_PIN is set to FAN_PIN when it's not. The "official" N3 firmware bypasses that line of the SanityCheck instead of fixing the underlying issue which won't fly with the actual Marlin git.

ellensp commented:
"convinced that E0_AUTO_FAN_PIN is set to FAN_PIN when it's not. "
This was a bug, fixed in https://github.com/MarlinFirmware/Marlin/pull/24593/files

mlee12382 commented:
> "convinced that E0_AUTO_FAN_PIN is set to FAN_PIN when it's not. "
> This was a bug, fixed in https://github.com/MarlinFirmware/Marlin/pull/24593/files 

I will update my marlin source and try again, I did update fairly recently though and it was still an issue,  but will try again just in case.

mlee12382 commented:
> "convinced that E0_AUTO_FAN_PIN is set to FAN_PIN when it's not. " This was a bug, fixed in https://github.com/MarlinFirmware/Marlin/pull/24593/files

Unfortunately even updating to the current 2.1.x bugfix files I am still getting the same error. :(
